distribution/config.go

```diff
@@ -86,7 +86,7 @@ type ImagePushConfig struct {
 type ImageConfigStore interface {
 	Put([]byte) (digest.Digest, error)
 	Get(digest.Digest) ([]byte, error)
-	RootFSAndOSFromConfig([]byte) (*image.RootFS, string, error)
+	RootFSOSOSVersionFromConfig([]byte) (*image.RootFS, string, string, error)
```

Comment:
Naming is hard

Reply:
`DecodeConfig` and return a struct with said config?

Reply:
Yes, was also not too happy with four separate values being returned